scipy/special/_logsumexp.py

```diff
@@ -212,4 +212,65 @@ def softmax(x, axis=None):
     """
 
     # compute in log space for numerical stability
-    return np.exp(x - logsumexp(x, axis=axis, keepdims=True))
+    # subtraction can cause cancellation when x and logsumexp(x) are close to each other.
+    # return np.exp(x - logsumexp(x, axis=axis, keepdims=True))
+    return np.exp(log_softmax(x, axis=axis))
```

Comment:
It would be great if you could add a test to the `softmax` tests that fails on `master` but passes on this PR due to the improved precision

Reply:
According to the Higham paper that @rlucas7 posted in the mailing list (https://arxiv.org/pdf/1909.03469.pdf) we should actually be computing softmax in a different way. (And it's Nick Higham, so he's almost surely right.)

Reply:
Since this PR is about log_softmax, I should have kept softmax as it was. I will revert the change and raise a new issue regarding the accuracy of softmax function.

Reply:
Awesome, that sounds good.